enterprise/tools/replay_action/replay_action.go

```diff
@@ -255,34 +266,49 @@ func contextWithTargetCredentials(ctx context.Context) context.Context {
 		var err error
 		ctx, err = setClientIdentity(ctx, targetClientIDKey)
 		if err != nil {
-			log.Fatalf("Failed to set target client identity: %s", err)
+			return ctx, fmt.Errorf("set target client identity: %s", err)
 		}
 	}
-	return ctx
+	return ctx, nil
 }
 
 func main() {
+	if err := run(); err != nil {
+		log.Fatal(err.Error())
+	}
+}
+
+func run() (err error) {
```

Comment:
any way to refactor this function so it's a little bit smaller and easier to understand?

Reply:
Good call, this function is getting too long. Decomposed into smaller functions and added comments. Might be a bit easier to review with whitespace enabled if you want to see the changes